itou/www/employee_record_views/views.py

```diff
@@ -83,12 +100,22 @@ def list_employee_records(request, template_name="employee_record/list.html"):
     )
     employee_record_badges = {row["status"]: row["cnt"] for row in employee_record_statuses}
 
-    eligibible_job_applications = JobApplication.objects.eligible_as_employee_record(siae)
+    eligibible_job_applications = (
+        JobApplication.objects.eligible_as_employee_record(siae)
+        .annotate(
+            has_suspension=Exists(Suspension.objects.filter(siae=siae, approval__jobapplication__pk=OuterRef("pk"))),
+            has_prolongation=Exists(
+                Prolongation.objects.filter(declared_by_siae=siae, approval__jobapplication__pk=OuterRef("pk"))
+            ),
+        )
+        .order_by(*job_application_order_by)
+    )
 
     # Set count of each status for badge display
     status_badges = [
         (
-            eligibible_job_applications.count(),
+            # Don't use count() since we also need the list when status is NEW
+            len(eligibible_job_applications),
```

Comment:
Mais quand on est pas sur les "NEW", on se retrouve à faire une requête SQL plus compliqué que nécessaire.
Cela ne ralentit pas trop les autres pages ?

Reply:
@francoisfreitag tu trouves que ce serait mieux avec la condition sutatus!=NEW , ou que mon explication te fais dire que c'est bien comme c'est actuellement ?

Reply:
J'aurais imaginé un truc du genre:
```
eligible_job_applications = JobApplication.objects.eligible_as_employee_record(siae)
if status == Status.NEW:
   # When showing NEW job applications, we need more infos
    eligible_job_applications = eligible_job_applications.annotate(
        has_suspension=Exists(Suspension.objects.filter(siae=siae, approval__jobapplication__pk=OuterRef("pk"))),
        has_prolongation=Exists(
            Prolongation.objects.filter(declared_by_siae=siae, approval__jobapplication__pk=OuterRef("pk"))
        ),
    )
    .order_by(*job_application_order_by)

# Set count of each status for badge display
status_badges = [
    (
        # Don't use count() since we also need the list when status is NEW
        len(eligible_job_applications) if status == Status.NEW else eligible_job_applications.count(),
...
```
qui ne me semble pas si compliqué mais c'est ptet effectivement de l'optimisation inutile :)